ring/src/piop/params.rs

```diff
@@ -30,8 +30,7 @@ pub struct PiopParams<F: PrimeField, P: AffineRepr<BaseField = F>> {
 }
 
 impl<F: PrimeField, P: AffineRepr<BaseField = F>> PiopParams<F, P> {
-    pub fn setup(domain: Domain<F>, h: P, seed: P) -> Self {
-        let padding_point = crate::hash_to_curve::<F, P>(b"w3f/ring-proof/padding");
+    pub fn setup(domain: Domain<F>, h: P, seed: P, padding_point: P) -> Self {
```

Comment:
maybe we could check if the points have the properties they need to have (on the curve, in the group, etc).